lib/codecs/src/decoding/format/protobuf.rs

```
@@ -0,0 +1,311 @@
+        )?;
+        let mut event = Event::Log(LogEvent::from(proto_vrl));
+        let event = match log_namespace {
+            LogNamespace::Vector => event,
+            LogNamespace::Legacy => {
+                let timestamp = Utc::now();
+
+                if let Some(timestamp_key) = log_schema().timestamp_key() {
+                    let log = event.as_mut_log();
+                    if !log.contains((PathPrefix::Event, timestamp_key)) {
+                        log.insert((PathPrefix::Event, timestamp_key), timestamp);
+                    }
+                }
+
+                event
+            }
+        };
+
+        Ok(smallvec![event])
+    }
+}
+
+impl From<&ProtobufDeserializerConfig> for ProtobufDeserializer {
+    fn from(config: &ProtobufDeserializerConfig) -> Self {
+        Self::new(config.desc_file.clone(), config.message_type.clone())
+    }
+}
+
+fn to_vrl(
+    prost_reflect_value: prost_reflect::Value,
+    kind: &prost_reflect::Kind,
+) -> vector_common::Result<vrl::value::Value> {
+    let v = match prost_reflect_value {
+        prost_reflect::Value::Bool(v) => vrl::value::Value::from(v),
+        prost_reflect::Value::I32(v) => vrl::value::Value::from(v),
+        prost_reflect::Value::I64(v) => vrl::value::Value::from(v),
+        prost_reflect::Value::U32(v) => vrl::value::Value::from(v),
+        prost_reflect::Value::U64(v) => vrl::value::Value::from(v),
+        prost_reflect::Value::F32(v) => vrl::value::Value::Float(
+            NotNan::new(f64::from(v)).map_err(|_e| format!("Float number cannot be Nan"))?,
+        ),
+        prost_reflect::Value::F64(v) => vrl::value::Value::Float(
+            NotNan::new(v).map_err(|_e| format!("F64 number cannot be Nan"))?,
+        ),
+        prost_reflect::Value::String(v) => vrl::value::Value::from(v),
+        prost_reflect::Value::Bytes(v) => vrl::value::Value::from(v),
+        prost_reflect::Value::EnumNumber(v) => {
+            let enum_desc = kind.as_enum().unwrap();
```

Let's return an error here instead of potentially panicking.